src/main/java/org/killbill/billing/plugin/adyen/dao/AdyenDao.java

```diff
@@ -294,8 +248,11 @@ public AdyenResponsesRecord withConnection(final Connection conn) throws SQLExce
                         : null,
                     toLocalDateTime(DateTime.now()),
                     tenantId.toString())
-                .returning()
-                .fetchOne();
+                .execute();
+
+            final long lastId = dslContext.lastID().longValue();
+            return dslContext.fetchOne(
+                ADYEN_RESPONSES, ADYEN_RESPONSES.RECORD_ID.eq(ULong.valueOf(lastId)));
```

Comment:
Instead of hardcoding `ULong`, maybe use `convert` to make it more generic.

Example here: https://github.com/killbill/killbill-stripe-plugin/blob/ec31abcfa6ef3a95a877ce05d6f9db370c17471d/src/main/java/org/killbill/billing/plugin/stripe/dao/StripeDao.java#L229

Reply:
Fixed